Walkthroughs/DockerDebugRemote/README.md

```diff
@@ -0,0 +1,203 @@
+ Version:           24.0.7
+ API version:       1.43
+ Go version:        go1.20.10
+ Git commit:        afdd53b
+ Built:             Thu Oct 26 09:07:51 2023
+ OS/Arch:           linux/amd64
+ Context:           default
+
+Server: Docker Engine - Community
+ Engine:
+  Version:          24.0.7
+  API version:      1.43 (minimum version 1.12)
+  Go version:       go1.20.10
+  Git commit:       311b9ff
+  Built:            Thu Oct 26 09:07:51 2023
+  OS/Arch:          linux/amd64
+  Experimental:     false
+ containerd:
+  Version:          1.6.24
+  GitCommit:        61f9fd88f79f081d64d6fa3bb1a0dc71ec870523
+ runc:
+  Version:          1.1.9
+  GitCommit:        v1.1.9-0-gccaecfc
+ docker-init:
+  Version:          0.19.0
+  GitCommit:        de40ad0
+```
+
+### 2. Add a new docker context for that system.
+
+```shell
+$ uname -a
+Darwin TDRYQ6WR2R 22.6.0 Darwin Kernel Version 22.6.0: Fri Sep 15 13:41:28 PDT 2023; root:xnu-8796.141.3.700.8~1/RELEASE_ARM64_T6020 arm64
+
+$ docker context create dock01 --docker "host=ssh://root@192.168.212.57"
+
+dock01
+Successfully created context "dock01"
+```
+
+### 3. Select the context; note that you can switch contexts back to your default if you no longer want to be administering the remote machine.
+
+```shell
+$ docker context use dock01
+dock01
+Current context is now "dock01"
+```
+
+### 4. Validate the context.
+
+```shell
+$ docker context ls
+NAME                TYPE                DESCRIPTION                               DOCKER ENDPOINT                                  KUBERNETES ENDPOINT   ORCHESTRATOR
+default             moby                Current DOCKER_HOST based configuration   unix:///var/run/docker.sock        
+desktop-linux       moby                Docker Desktop                            unix:///Users/jschmidt/.docker/run/docker.sock
+dock01 *            moby                                                          ssh://root@192.168.212.57          
+```
+
+### 5. Check the context; you can see that we are using the remote docker installation (note the difference in arch between the client and server).
+
+```shell
+$ docker version
+Client:
+ Cloud integration: v1.0.35+desktop.5
+ Version:           24.0.6
+ API version:       1.43
+ Go version:        go1.20.7
+ Git commit:        ed223bc
+ Built:             Mon Sep  4 12:28:49 2023
+ OS/Arch:           darwin/arm64
+ Context:           dock01
+
+Server: Docker Engine - Community
+ Engine:
+  Version:          24.0.7
+  API version:      1.43 (minimum version 1.12)
+  Go version:       go1.20.10
+  Git commit:       311b9ff
+  Built:            Thu Oct 26 09:07:51 2023
+  OS/Arch:          linux/amd64
+  Experimental:     false
+ containerd:
+  Version:          1.6.24
+  GitCommit:        61f9fd88f79f081d64d6fa3bb1a0dc71ec870523
+ runc:
+  Version:          1.1.9
+  GitCommit:        v1.1.9-0-gccaecfc
+ docker-init:
+  Version:          0.19.0
+  GitCommit:        de40ad0
+  ```
+  
+ ### 6. Run an image on the remote system.
```

Comment:
Should we mention the case where the container we are interested in debugging is already running in this remote environment? Since we won't need to run a new image in that case.